habitat_sim/simulator.py

```diff
@@ -341,16 +322,17 @@ def get_sensor_observations(
             return_single = False
 
         for agent_id in agent_ids:
-            agent_sensorsuite = self.__sensors[agent_id]
-            for _sensor_uuid, sensor in agent_sensorsuite.items():
-                sensor.draw_observation()
+            for uuid in self.get_agent(agent_id).scene_node.node_sensors.items():
+                self.draw_observation(self._sensor_readers[uuid])
```

Comment:
May I understand the purpose to build a simulator level "draw_observation"? Such usage seems unnatural to me.
I think it still makes sense to call `sensor.draw_observation()`. The only difference is that where the sensors come from. Before, a sensor is from the sensorSuite in the Agent class. Now it should be from the node where the Agent resides in.

Reply:
Yes, I think it makes sense that the sensor should draw the observation as well. Sorry, from earlier feedback that the simulator should drive the drawing process, I was under the impression that simulator should own the function to draw the observation, while SensorReader would just own the implementation to initialize the noise model and buffers. https://github.com/facebookresearch/habitat-sim/pull/1118#discussion_r590851442